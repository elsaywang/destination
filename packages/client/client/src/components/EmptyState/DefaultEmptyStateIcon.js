import React from 'react';

export default () => (
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 140.123 104.287" width="100">
        <defs>
            <style
                dangerouslySetInnerHTML={{
                    __html:
                        '.cls-1,.cls-2{fill:none;stroke:#b3b3b3;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:10}.cls-1{stroke-width:3px}.cls-2{stroke-width:2px}',
                }}
            />
        </defs>
        <g id="Group_20121" data-name="Group 20121" transform="translate(-57 -.065)">
            <path
                id="Line_4688"
                data-name="Line 4688"
                className="cls-1"
                transform="translate(97.3 22)"
                d="M30.3 0L0 12.7"
            />
            <path
                id="Path_7879"
                data-name="Path 7879"
                className="cls-1"
                d="M127.6 22L181 44.3l14.2-18.8a1.907 1.907 0 0 0-.9-3l-50-20.7a1.917 1.917 0 0 0-2.4.7L127.6 22"
            />
            <path
                id="Path_7880"
                data-name="Path 7880"
                className="cls-1"
                d="M73.2 44.3l-14.7-20 54 22.4 14.9 19.7 53.8-22.1v34.9a2.241 2.241 0 0 1-1.2 1.9l-52 21.6a1.95 1.95 0 0 1-1.5 0L74.4 81a1.975 1.975 0 0 1-1.2-1.8V44.7"
            />
            <path
                id="Line_4689"
                data-name="Line 4689"
                className="cls-2"
                transform="translate(127.7 72.2)"
                d="M0 0v23"
            />
            <path
                id="Line_4690"
                data-name="Line 4690"
                className="cls-2"
                transform="translate(81.8 48.5)"
                d="M0 0l35 14.9"
            />
        </g>
    </svg>
);
