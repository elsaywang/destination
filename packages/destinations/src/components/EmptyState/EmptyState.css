:local(.center) {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

:local(.content) {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top:3.5rem;
}

:local(.primaryMessage) {
    color: var(--spectrum-heading-subtitle2-text-color-default);
    font-size: var(--spectrum-global-font-size-xlarge);
    font-weight: var(--spectrum-global-font-weight-light);
    padding-top: 0.95rem;
  }

:local(.secondaryMessage) {
    color: var(--spectrum-label-text-color-disabled);
    font-size: var(--spectrum-global-font-size-medium);
    font-weight: var(--spectrum-global-font-weight-regular);
    font-style: italic;
    margin-top: -0.75rem;
  }
