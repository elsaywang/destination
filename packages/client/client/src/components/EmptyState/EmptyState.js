import React from 'react';
import PropTypes from 'prop-types';
import styles from './EmptyState.css';
import DefaultEmptyStateIcon from './DefaultEmptyStateIcon';

const EmptyState = ({ primaryMessage, secondaryMessage, children, icon: Icon }) => (
    <div className={styles.content} data-test="empty-state">
        <Icon size="XXL" />
        <PrimaryMessage message={primaryMessage} />
        {secondaryMessage && <SecondaryMessage message={secondaryMessage} />}
        {children && { children }}
    </div>
);

const PrimaryMessage = ({ message }) => <div className={styles.primaryMessage}>{message}</div>;

const SecondaryMessage = ({ message }) => (
    <div className={styles.secondaryMessage}>
        <p>{message}</p>
    </div>
);


EmptyState.propTypes = {
    primaryMessage: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
    secondaryMessage: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    icon: PropTypes.func,
};

EmptyState.defaultProps = {
    secondaryMessage: null,
    icon: DefaultEmptyStateIcon,
};

export { PrimaryMessage, SecondaryMessage };
export default EmptyState;
