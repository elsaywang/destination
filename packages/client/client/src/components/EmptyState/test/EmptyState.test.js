import React from 'react';
import { shallow } from 'enzyme';
import EmptyState, { PrimaryMessage, SecondaryMessage } from '../EmptyState';
import DefaultEmptyStateIcon from '../DefaultEmptyStateIcon';

describe('rendering', () => {
    const props = {
        primaryMessage: 'Start adding accounts to enable people based destinations',
        secondaryMessage: '',
    };
    const wrapper = shallow(<EmptyState {...props} />);

    it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render <DefaultEmptyStateIcon/> as default icon', () => {
        expect(wrapper.find(DefaultEmptyStateIcon)).toBeTruthy();
    });
    
    it('should always contain <PrimaryMessage/>', () => {
        expect(wrapper.find(PrimaryMessage)).toBeTruthy();
    });

    it('should <SecondaryMessage/> as  `secondaryMessage` props has no value', () => {
        expect(wrapper.find(SecondaryMessage)).toBeTruthy();
    });
});
