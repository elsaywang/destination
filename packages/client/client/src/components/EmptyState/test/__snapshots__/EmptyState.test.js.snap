// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmptyState
    icon={[Function]}
    primaryMessage="Start adding accounts to enable people based destinations"
    secondaryMessage=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          size="XXL"
/>,
        <PrimaryMessage
          message="Start adding accounts to enable people based destinations"
/>,
        "",
        undefined,
      ],
      "className": undefined,
      "data-test": "empty-state",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "size": "XXL",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "message": "Start adding accounts to enable people based destinations",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "",
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            size="XXL"
/>,
          <PrimaryMessage
            message="Start adding accounts to enable people based destinations"
/>,
          "",
          undefined,
        ],
        "className": undefined,
        "data-test": "empty-state",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": "XXL",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "message": "Start adding accounts to enable people based destinations",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
